### ISSUE TITLE
NLP: Implement training loops for classification, NER, and embeddings

### ISSUE THREAD

**Issue by kiyeonjeon21** [kiyeonjeon21/latent-lab#1]
## Context

The NLP domain modules (`src/latent_lab/nlp/`) currently load pretrained HuggingFace models but don't implement actual training or fine-tuning.

## Current state

| Module | LOC | Status |
|--------|-----|--------|
| `classification.py` | 24 | Loads pretrained model, no training loop |
| `ner.py` | 25 | Loads NER pipeline, no fine-tuning |
| `embeddings.py` | ~30 | Wrapper around sentence-transformers |
| `tokenizer.py` | 18 | Tokenizer analysis only (done) |

## What needs to happen

- [ ] `classification.py` — Add fine-tuning loop with HuggingFace Trainer on a text classification dataset (e.g., SST-2, IMDB). Log accuracy/F1 to MLflow.
- [ ] `ner.py` — Add fine-tuning on a NER dataset (e.g., CoNLL-2003). Log per-entity F1.
- [ ] `embeddings.py` — Add embedding quality evaluation (e.g., similarity benchmarks or clustering visualization).

## Interface

Every module must expose `run(cfg: DictConfig) -> None` and use the existing MLflow tracker:

```python
from latent_lab.experiments.tracker import log_config, log_metrics, setup_tracking, track_run

def run(cfg: DictConfig) -> None:
    setup_tracking(f"nlp-{cfg.name}")
    with track_run(run_name=cfg.name, tags={"domain": "nlp", "task": "classification"}):
        log_config(cfg)
        # training logic here
        log_metrics({"accuracy": acc, "f1": f1})
```

## Notes

- Target device: MPS GPU (`torch.device("mps")`)
- Keep datasets small enough to run on 32GB unified memory
- Add a matching config in `configs/experiments/` if one doesn't exist
